File: go-api/models/teacher.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Teacher struct{
	ID primitive.ObjectID `json:"id" bson:"_id"`
	Name string `json:"name" bson:"name" validate:"required"`
	Email string `json:"email" bson:"email" validate:"required"`
	DOB string `json:"dob" bson:"dob" validate:"required"`
	TeachingCourses []Course `json:"teachingCourses" bson:"teachingCourses"`
	TeachingClasses []TeachingClasses `json:"teachingClasses" bson:"teachingClasses"`
}

type TeachingClasses struct{
	ID primitive.ObjectID `json:"id" bson:"_id" validate:"required"`
	Name string `json:"name" bson:"name" validate:"required"`
}

File: client/src/types/teacher.ts
import { Course } from "./course";

export interface Teacher {
  id: string;
  name: string;
  email: string;
  dob: string;
  teachingCourses: Course[];
  teachingClasses: TeachingClasses[];
}

export interface TeachingClasses {
  id: string;
  name: string;
}

export interface GetTeachersResponse {
  data: Teacher[];
}

export interface AddTeacherResponse {
  result: Teacher;
}
